conformance/lang/expressions/new-expr/explicit-new-expr/stream_class_descriptor.balt

```diff
@@ -0,0 +1,303 @@
+Test-Case: error
+Description: Test new-expression to create a stream without stream type parameters.
+Labels: explicit-new-expr, stream-type, var
+
+function init() {
+    var _ = new stream<>(); // @error missing type parameters
+}
+
+Test-Case: error
+Description: Test new-expression to create a stream with incompatible stream type parameters.
+Labels: new-expr, stream-type-descriptor
+
+function init() {
+    stream<int> _ = new stream<string>(); // @error incompatible types: expected 'stream<int>', found 'stream<string>'
+}
+
+Test-Case: output
+Description: Test new-expression to create a stream with incompatible type-descriptor.
+Labels: error, optional-type, new-expr, string, stream-type-descriptor
+
+function init() {
+    stream s = new stream<string, error?>();
+    io:println(s); // @output stream <stream<string,error?>>
+}
+
+Test-Case: output
+Description: Test new-expression to create a empty streams.
+Labels: int, record-type, optional-type, is-expr, module-type-defn, new-expr, union-type, stream-type-descriptor
+
+type IntRecord record {|
+    int value;
+|};
+
+function init() {
+    stream<int, error?> emptyStream = new stream<int, error?>();
+    IntRecord|error? d = emptyStream.next();
+    io:println(d is ()); // @output true
+}
+
+Test-Case: output
+Description: Test new-expr to create a stream without completion type.
+Labels: int, record-type, optional-type, new-expr, is-expr, stream-type-descriptor
+
+type IntRecord record {|
+    int value;
+|};
+
+function init() {
+    stream<int> emptyStream = new stream<int>();
+    IntRecord? d = emptyStream.next();
+    io:println(d is ()); // @output true
+}
+
+Test-Case: error
+Description: Test new-expr to create a stream with stream extra parameters.
+Labels: int, error, optional-type, new-expr, module-class-defn, stream-type-descriptor
+
+class NumbersGenerator {
+    int i = 0;
+    public isolated function next() returns string|error? {
+        self.i += 1;
+        return self.i.toBalString();
+    }
+}
+
+function init() {
+    NumbersGenerator numbersGenerator = new;
+    stream<string, error?> numberStream = new stream<string, error?>(numbersGenerator, string); // @error 'new(numbersGenerator, null)' is not a valid constructor for streams type
+}
+
+
+Test-Case: error
+Description: Test new-expr to create a stream with stream implementor with next method with incompatible return type.
+Labels: int, error, optional-type, new-expr, module-class-defn, stream-type-descriptor
```

Comment:
`return-stmt`, `isolated-function` labels missing.

Reply:
Thanks no need to add `return-stmt` as in minimal subset and not testing it specifically here.